clang/test/SemaSYCL/sycl-device.cpp

```diff
@@ -6,50 +6,47 @@
-void bar() {}
-
-__attribute__((sycl_device)) // expected-error {{'sycl_device' attribute cannot be applied to a static function or function in an anonymous namespace}}
+__attribute__((sycl_device(0))) // expected-error {{'sycl_device' attribute cannot be applied to a static function or function in an anonymous namespace}}
 static void func1() {}
 
 namespace {
-  __attribute__((sycl_device)) // expected-error {{'sycl_device' attribute cannot be applied to a static function or function in an anonymous namespace}}
+  __attribute__((sycl_device(0))) // expected-error {{'sycl_device' attribute cannot be applied to a static function or function in an anonymous namespace}}
   void func2() {}
 }
 
 class A {
-  __attribute__((sycl_device)) // expected-error {{'sycl_device' attribute cannot be applied to a class member function}}
+  __attribute__((sycl_device(0))) // expected-error {{'sycl_device' attribute cannot be applied to a class member function}}
   A() {}
 
-  __attribute__((sycl_device)) // expected-error {{'sycl_device' attribute cannot be applied to a class member function}}
+  __attribute__((sycl_device(0))) // expected-error {{'sycl_device' attribute cannot be applied to a class member function}}
   int func3() {}
 };
 
 #if defined(NOT_STRICT)
-__attribute__((sycl_device))
+__attribute__((sycl_device(0)))
 int* func3() { return nullptr; }
 
-__attribute__((sycl_device))
+__attribute__((sycl_device(0)))
 void func3(int *) {}
 #elif defined(WARN_STRICT)
-__attribute__((sycl_device)) // expected-warning {{SYCL 1.2.1 specification does not allow 'sycl_device' attribute applied to a function with a raw pointer return type}}
+__attribute__((sycl_device(0))) // expected-warning {{SYCL 1.2.1 specification does not allow 'sycl_device' attribute applied to a function with a raw pointer return type}}
 int* func3() { return nullptr; }
 
-__attribute__((sycl_device)) // expected-warning {{SYCL 1.2.1 specification does not allow 'sycl_device' attribute applied to a function with a raw pointer parameter type}}
+__attribute__((sycl_device(0))) // expected-warning {{SYCL 1.2.1 specification does not allow 'sycl_device' attribute applied to a function with a raw pointer parameter type}}
 void func3(int *) {}
 #else
-__attribute__((sycl_device)) // expected-error {{SYCL 1.2.1 specification does not allow 'sycl_device' attribute applied to a function with a raw pointer return type}}
+__attribute__((sycl_device(0))) // expected-error {{SYCL 1.2.1 specification does not allow 'sycl_device' attribute applied to a function with a raw pointer return type}}
 int* func3() { return nullptr; }
 
-__attribute__((sycl_device)) // expected-error {{SYCL 1.2.1 specification does not allow 'sycl_device' attribute applied to a function with a raw pointer parameter type}}
+__attribute__((sycl_device(0))) // expected-error {{SYCL 1.2.1 specification does not allow 'sycl_device' attribute applied to a function with a raw pointer parameter type}}
```

Comment:
Could you please add a compiler-specific test case for `sycl_device` attribute with value `1`, i.e. check that raw-pointer related diagnostic is not emitted when value `1` is used?

Reply:
Done